packages/shared-ini-file-loader/src/slurpFile.ts

```diff
@@ -0,0 +1,62 @@
+// ToDo: Change to "fs/promises" when supporting nodejs>=14
+import { promises as fsPromises } from "fs";
+
+const { readFile, stat } = fsPromises;
+
+type callbacks = { resolve: Function; reject: Function };
+type fileStatus = {
+  contents: string;
+  isReading: boolean;
+  lastModified: Date;
+  requestQueue: callbacks[];
+};
+
+const fileReadHash: { [key: string]: fileStatus } = {};
+
+const callReadFile = (path: string, cbs: callbacks) => {
+  fileReadHash[path].requestQueue.push(cbs);
+  readFile(path, "utf8")
+    .then((data) => {
+      // File read successful
+      fileReadHash[path].contents = data;
+      fileReadHash[path].requestQueue.forEach(({ resolve }) => resolve(data));
+    })
+    .catch((err) => {
+      fileReadHash[path].requestQueue.forEach(({ reject }) => reject(err));
+    })
+    .finally(() => {
+      fileReadHash[path].isReading = false;
+      fileReadHash[path].requestQueue = [];
+      // Update last modified.
+      stat(path).then(({ mtime }) => {
+        fileReadHash[path].lastModified = mtime;
+      });
+    });
+};
+
+export const slurpFile = (path: string) =>
```

Comment:
I think the implemention doesn't need to be this complecated. Why don't we implement the promise call back queue for the `slurpFile()` (or even `loadSharedConfigFiles()`) instead of the `readFile`? So that we don't need to worry about file reading status or last modified time.

Reply:
yes, that was exactly what I was thinking here: https://github.com/aws/aws-sdk-js-v3/pull/3281/files#r799850967
👍

Reply:
Superceded by promise callback queue for `loadSharedConfigFiles()` in https://github.com/aws/aws-sdk-js-v3/pull/3282